fitparser/src/de/parser.rs

```diff
@@ -552,27 +499,31 @@ fn data_message_fields_impl<'a>(
             field_def.size,
         )?;
         if let Some(value) = value {
-            fields.insert(field_def.field_definition_number, value);
+            fields.insert((255u8, field_def.field_definition_number), value);
         }
         input = i;
     }
-    // store developer data as a byte array since we don't handle these fields in the final data
-    // output yet
-    let mut developer_fields = Vec::new();
+    // handle developer fields analogously to fixed fields
     for field_def in &def_mesg.developer_field_definitions {
+        let dev_field_description = developer_fields
+            .get(&(field_def.developer_data_index, field_def.field_number))
+            .expect("Dev field must previously be defined");
```

Comment:
Instead of calling `.expect(...)` here lets try to hook in a new `ErrorKind` variant to avoid the panic and make it something recoverable downstream.

Reply:
I tried creating a new ErrorKind, but I was not happy with the complexity that added to hand it through the IResult functions. Instead for now I have just used a standard nom Error.

Reply:
Ah you're right, I didn't realize where this was at in the code when I made the comment. Using a standard nom error is the way to go here.